js_modules/dagster-ui/packages/ui-core/src/overview/OverviewSchedules.tsx

```diff
@@ -0,0 +1,372 @@
+      };
+    });
+  }, [repoBuckets, runningState]);
+
+  const sanitizedSearch = searchValue.trim().toLocaleLowerCase();
+  const anySearch = sanitizedSearch.length > 0;
+
+  const filteredBySearch = useMemo(() => {
+    const searchToLower = sanitizedSearch.toLocaleLowerCase();
+    return filteredBuckets
+      .map(({repoAddress, schedules}) => ({
+        repoAddress,
+        schedules: schedules.filter(({name}) => name.toLocaleLowerCase().includes(searchToLower)),
+      }))
+      .filter(({schedules}) => schedules.length > 0);
+  }, [filteredBuckets, sanitizedSearch]);
+
+  const anySchedulesVisible = useMemo(
+    () => filteredBySearch.some(({schedules}) => schedules.length > 0),
+    [filteredBySearch],
+  );
+
+  // Collect all schedules across visible code locations that the viewer has permission
+  // to start or stop.
+  const allPermissionedSchedules = useMemo(() => {
+    return repoBuckets
+      .map(({repoAddress, schedules}) => {
+        return schedules
+          .filter(({scheduleState}) => filterPermissionedInstigationState(scheduleState))
+          .map(({name, scheduleState}) => ({
+            repoAddress,
+            scheduleName: name,
+            scheduleState,
+          }));
+      })
+      .flat();
+  }, [repoBuckets]);
+
+  // Build a list of keys from the permissioned schedules for use in checkbox state.
+  // This includes collapsed code locations.
+  const allPermissionedScheduleKeys = useMemo(() => {
+    return allPermissionedSchedules.map(({repoAddress, scheduleName}) =>
+      makeScheduleKey(repoAddress, scheduleName),
+    );
+  }, [allPermissionedSchedules]);
+
+  const [{checkedIds: checkedKeys}, {onToggleFactory, onToggleAll}] = useSelectionReducer(
+    allPermissionedScheduleKeys,
+  );
+
+  // Filter to find keys that are visible given any text search.
+  const permissionedKeysOnScreen = useMemo(() => {
+    const filteredKeys = new Set(
+      filteredBySearch
+        .map(({repoAddress, schedules}) => {
+          return schedules.map(({name}) => makeScheduleKey(repoAddress, name));
+        })
+        .flat(),
+    );
+    return allPermissionedScheduleKeys.filter((key) => filteredKeys.has(key));
+  }, [allPermissionedScheduleKeys, filteredBySearch]);
+
+  // Determine the list of schedule objects that have been checked by the viewer.
+  // These are the schedules that will be operated on by the bulk start/stop action.
+  const checkedSchedules = useMemo(() => {
+    const checkedKeysOnScreen = new Set(
+      permissionedKeysOnScreen.filter((key: string) => checkedKeys.has(key)),
+    );
+    return allPermissionedSchedules.filter(({repoAddress, scheduleName}) => {
+      return checkedKeysOnScreen.has(makeScheduleKey(repoAddress, scheduleName));
+    });
+  }, [permissionedKeysOnScreen, allPermissionedSchedules, checkedKeys]);
+
+  const viewerHasAnyInstigationPermission = allPermissionedScheduleKeys.length > 0;
+  const checkedCount = checkedSchedules.length;
+
```

Comment:
There is a lot going on in here! Thanks for writing super nice + clear variable names the whole way through 😅 it's a lot but very easy to follow